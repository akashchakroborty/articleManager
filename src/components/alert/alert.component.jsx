import React from "react";
import { connect } from "react-redux";
import { AlertContainer } from "./alert.styles";

export const Alert = ({ alerts }) =>
  alerts !== null &&
  alerts.length > 0 &&
  alerts.map((alert) => (
    <AlertContainer
      className="alert-message"
      key={alert.id}
      className={`alert alert-${alert.alertType}`}
    >
      {alert.msg}
    </AlertContainer>
  ));

const mapStateToProps = (state) => ({
  alerts: state.alerts,
});

export default connect(mapStateToProps)(React.memo(Alert));
