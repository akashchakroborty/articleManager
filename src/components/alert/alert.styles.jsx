import styled from "styled-components";

const darkColor = "#343a40";
const lightColor = "#f4f4f4";
const dangerColor = "#dc3545";
const successColor = "#28a745";
const primaryColor = "#17a2b8";

export const AlertContainer = styled.div`
  width: 850px;
  display: flex;
  justify-content: center;
  margin: 30px auto;
  @media screen and (max-width: 800px) {
    width: unset;
  }
  &.alert {
    padding: 0.8rem;
    opacity: 0.9;
    background: ${primaryColor};
    color: #333;
  }

  &.alert-primary {
    background: ${primaryColor};
    color: #fff;
  }

  &.alert-light {
    background: ${lightColor};
    color: #333;
  }

  .alert-dark {
    background: ${darkColor};
    color: #fff;
  }

  &.alert-danger {
    background: ${dangerColor};
    color: #fff;
  }

  &.alert-success {
    background: ${successColor};
    color: #fff;
  }

  &.alert-white {
    background: #fff;
    color: #333;
    border: #ccc solid 1px;
  }
`;

AlertContainer.displayName = "AlertContainer";
