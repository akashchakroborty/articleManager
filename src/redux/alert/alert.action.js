import { v4 as uuid } from "uuid";
import { SET_ALERT, REMOVE_ALERT } from "./alert.types";

export const setAlert = (msg, alertType, timeout = 3000, alertId) => async (
  dispatch
) => {
  const id = alertId || uuid();
  dispatch({
    type: SET_ALERT,
    payload: { msg, alertType, id },
  });
  setTimeout(
    () =>
      dispatch({
        type: REMOVE_ALERT,
        payload: id,
      }),
    timeout
  );
};
