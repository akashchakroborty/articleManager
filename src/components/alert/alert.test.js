import React from "react";
import { shallow } from "enzyme";

import { Alert } from "./alert.component";

describe("Alert Component", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      alerts: [],
    };

    wrapper = shallow(<Alert {...mockProps} />);
  });

  it("should render Alert component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("if alerts is empty", () => {
    it("should not render AlertContainer component", () => {
      expect(wrapper.exists("AlertContainer")).toBe(false);
      expect(wrapper.find("AlertContainer").length).toEqual(0);
    });
  });

  describe("if alerts is not empty", () => {
    const newMockProps = {
      alerts: [{ msg: "mockAlert", alertType: "primary", id: 1 }],
    };

    const newWrapper = shallow(<Alert {...newMockProps} />);
    it("should render AlertContainer component", () => {
      expect(newWrapper.exists("AlertContainer")).toBe(true);
      expect(newWrapper.find("AlertContainer").length).toEqual(1);
      expect(newWrapper.find("AlertContainer").hasClass("alert-primary")).toBe(
        true
      );
      expect(newWrapper.find("AlertContainer").hasClass("alert")).toBe(true);
      expect(newWrapper.find("AlertContainer").text()).toBe("mockAlert");
    });
  });
});
