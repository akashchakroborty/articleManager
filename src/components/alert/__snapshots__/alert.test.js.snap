// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component should render Alert component 1`] = `ShallowWrapper {}`;
